packages/components/tooltip/src/use-tooltip.ts

```diff
@@ -170,6 +180,12 @@ export function useTooltip(originalProps: UseTooltipProps) {
     containerPadding,
   });
 
+  useLayoutEffect(() => {
+    if (!updatePositionDeps.length) return;
+    // force update position when deps change
+    updatePosition();
+  }, updatePositionDeps);
```

Comment:
This looks dangerous, are you sure this isn't going to have some edge cases where we are calling this a bunch of times and are gonna get performance issues?

Reply:
I don't think so, it all depends on the dependencies it will not affect production implementation because the dependencies are empty by default and for the Slider/Tooltip implementation we only call it whenever the thumb value changes